admin/services/pin-generation.service.js

```diff
@@ -88,19 +92,37 @@ pinGenerationService.isValid = async (p) => {
 /**
  * Generate pupils pins
  * @param pupilsList
+ * @param dfeNumber
  * @returns {Array}
  */
-pinGenerationService.updatePupilPins = async (pupilsList) => {
-  const ids = Object.values(pupilsList || null)
+pinGenerationService.updatePupilPins = async (pupilsList, dfeNumber) => {
+  let ids = Object.values(pupilsList || null)
```

Comment:
is pupilsList supposed to be an array? if so, state that in the jsDoc comments above and check for it with `Array.isArray`